types/index.d.ts

```diff
@@ -0,0 +1,491 @@
+    } & Registration.Callbacks;
+    // Feature flags
+    features?: {
+      [featureKey in Feature]?: boolean;
+    };
+    // OIDC
+    clientId?: string;
+    redirectUri?: string;
+    authParams?: AuthParams;
+    authClient?: OktaAuth;
+    idps?: Array<Idp | IdpLink>;
+    idpDisplay?: IdpDisplay;
+    oAuthTimeout?: number;
+    // Smart Card IdP (X509)
+    piv?: Piv;
+    // Bootstrapping from a recovery token
+    recoveryToken?: string;
+  }
+
+  // Auth params
+  interface AuthParams extends OktaAuthOptions {
+    display?: Display;
+    responseType?: ResponseType | Array<ResponseType>;
+    responseMode?: ResponseMode;
+    scopes?: Array<Scope>;
+    state?: string;
+    nonce?: string;
+    authScheme?: string;
+  }
+  type Display =
+    'popup' |
+    'page';
+  type ResponseMode =
+    'okta_post_message' |
+    'fragment' |
+    'query' |
+    'form_post';
+  type ResponseType =
+    'code' |
+    'token' |
+    'id_token' |
+    'access_token';
+  type Scope =
+    'openid' |
+    'email' |
+    'profile' |
+    'address' |
+    'phone';
+
+  // IdPs
+  interface Idp {
+    type: string;
+    id: string;
+  }
+  interface IdpLink {
+    text: string;
+    id: string;
+    className?: string;
+  }
+  type IdpDisplay =
+    'PRIMARY' |
+    'SECONDARY';
+  interface Piv {
+    certAuthUrl: string;
+    text?: string;
+    className?: string;
+    isCustomDomain?: boolean;
+  }
+
+  // Types for config
+  type UserOperation =
+    'PRIMARY_AUTH' |
+    'FORGOT_PASSWORD' |
+    'UNLOCK_ACCOUNT';
+
+  interface Creds {
+    username: string;
+    password: string;
+  }
+
+  type SimpleCallback = () => void;
+
+  type LanguageCode =
+    'cs' | // Czech
+    'da' | // Danish
+    'de' | // German
+    'el' | // Greek
+    'en' | // English
+    'es' | // Spanish
+    'fi' | // Finnish
+    'fr' | // French
+    'hu' | // Hungarian
+    'id' | // Indonesian
+    'it' | // Italian
+    'ja' | // Japanese
+    'ko' | // Korean
+    'ms' | // Malaysian
+    'nb' | // Norwegian
+    'nl-NL' | // Dutch
+    'pl' | // Polish
+    'pt-BR' | // Portuguese (Brazil)
+    'ro' | // Romanian
+    'ru' | // Russian
+    'sv' | // Swedish
+    'th' | // Thai
+    'tr' | // Turkish
+    'uk' | // Ukrainian
+    'zh-CN' | // Chinese (PRC)
+    'zh-TW'; // Chinese
+
+  type LanguageCallback = (supportedLanguages: Array<LanguageCode>, userLanguages: Array<string>) => LanguageCode;
+
+  type ColorKey = 'brand';
+
+  interface LinkBasic {
+    text: string;
+    href: string;
+  }
+  type LinkTarget =
+    '_blank' |
+    '_self' |
+    '_parent' |
+    '_top';
+  interface Link extends LinkBasic {
+    target?: LinkTarget;
+  }
+
+  interface CustomButtonBasic {
+    click: SimpleCallback;
+  }
+  interface CustomButton extends CustomButtonBasic {
+    title?: string;
+    i18nKey?: string;
+    className?: string;
+  }
+
+  interface FieldError {
+    errorSummary: string;
+    reason?: string;
+    resource?: string;
```

Comment:
How did you figure out those properties?
What I found are
```
private String errorSummary;
private String reason;
private String location;
private String locationType;
private String domain;
```

Reply:
I've found in readme: https://github.com/okta/okta-signin-widget#use-field-level-error
Changed